apiserver/stateauthenticator/context.go

```diff
@@ -319,7 +322,7 @@ func (ctxt *authContext) externalMacaroonAuth(ctx context.Context, identClient i
 // newExternalMacaroonAuth returns an authenticator that can authenticate
 // macaroon-based logins for external users. This is just a helper function
 // for authCtxt.externalMacaroonAuth.
-func newExternalMacaroonAuth(ctx context.Context, st *state.State, controllerConfigService ControllerConfigService, clock clock.Clock, expiryTime time.Duration, identClient identchecker.IdentityClient) (*authentication.ExternalMacaroonAuthenticator, error) {
+func newExternalMacaroonAuth(ctx context.Context, st *state.State, controllerConfigService ControllerConfigService, bakeryConfigService BakeryConfigService, clock clock.Clock, expiryTime time.Duration, identClient identchecker.IdentityClient) (*authentication.ExternalMacaroonAuthenticator, error) {
```

Comment:
Could the be replaced with a argument struct?

Reply:
Done